src/otx/core/exporter/base.py

```diff
@@ -280,30 +280,17 @@ def _postprocess_openvino_model(self, exported_model: openvino.Model) -> openvin
             # workaround for OVC's bug: single output doesn't have a name in OV model
             exported_model.outputs[0].tensor.set_names({"output1"})
 
+        # name assignment process is similar to torch onnx export
         if self.output_names is not None:
-            traced_outputs = [(output.get_names(), output) for output in exported_model.outputs]
-
-            for output_name in self.output_names:
-                found = False
-                for name, output in traced_outputs:
-                    # TODO(vinnamkim): This is because `name` in `traced_outputs` is a list of set such as
-                    # [{'logits', '1555'}, {'1556', 'preds'}, {'1557', 'scores'},
-                    # {'saliency_map', '1551'}, {'feature_vector', '1554', 'input.1767'}]
-                    # This ugly format of `name` comes from `openvino.convert_model`.
-                    # Find a cleaner way for this in the future.
-                    if output_name in name:
-                        found = True
-                        # NOTE: This is because without this renaming such as
-                        # `{'saliency_map', '1551'}` => `{'saliency_map'}`
-                        # ModelAPI cannot produce the outputs correctly.
-                        output.tensor.set_names({output_name})
-
-                if not found:
-                    msg = (
-                        "Given output name to export is not in the traced_outputs, "
-                        f"{output_name} not in {traced_outputs}"
-                    )
-                    raise RuntimeError(msg)
+            if len(exported_model.outputs) >= len(self.output_names):
+                for i, name in enumerate(self.output_names):
+                    exported_model.outputs[i].tensor.set_names({name})
```

Comment:
How do you guarantee that `exported_model.outputs[:len(self.output_names)]` always indicate `self.output_names` and their orders is same?

Reply:
Unfortunately, OVC API docs do not describe any properties of the output model, so formally nothing (including naming is guaranteed). I believe that happened because of multiple backends supported. From the past experience, output order is one of a few details that is preserved. Also, I additionally clarified that with OVC owner, he confirmed that the output order is guaranteed.

Reply:
The inconsistency between them can also be happened by human errors. For example, the model implementation produces the output sinks in this order: `output1`, `output2`, and `output3`, but given `output_names` in the exporter can be `["output2", "output1", "output3"]`. In this case, we do not have any exception raised, then the real problem is hidden and could not be found easily.

Reply:
I have encountered an error related to this - https://jira.devtools.intel.com/browse/CVS-138604.
I found that it is simply a difference in the order of the output.
We need to clean this up. As an external user, if it doesn't work just because the order of this is different, it's hard to find the problem with this. It seems like we need to write more documentation for now.

Reply:
Unfortunately, we don't have a fix for that on torch side as well. If we export a model via ONNX or to ONNX, incorrect order will be propagated to OV model as well, and the cleaning process we had before, can not catch that as well, because it has no cue about the correct output order. Here, I'm just aligning the behavior to avoid incorrect results when `via_onnx==False`.

From the top of my head, I can only propose to preprocess the torch model itself to avoid this, but we need an extra inference and a wrapper to reorder the outputs if required. Also, in the docs we can mention that the order is important.

Reply:
> Unfortunately, we don't have a fix for that on torch side as well. If we export a model via ONNX or to ONNX, incorrect order will be propagated to OV model, and the cleaning process we had before, can not catch that as well, because it has no cue about the correct output order. Here, I'm just aligning the behavior to avoid incorrect results when `via_onnx==False`.
> 
> From the top of my head, I can only propose to preprocess the torch model itself to avoid this, but we need an extra inference and a wrapper to reorder the outputs if required. Also, in the docs we can mention that the order is important.

Sorry, I don't understand what you mean. To me, your current implementation is hiding a restriction that the model output sink order and names should be same with `output_names`. In addition, even if the wrong `output_names` is given, it does not raise any exception and make any malfunction afterward hard to trace. However, the previous implementation at least tries to raise an error if not acceptable input is given.

What's the reason that this implementation is more acceptable than before?

Reply:
By design this feature doesn't suppose the raw model outputs have names. The exporter tries to make no difference between ONNX and OV, since we should support both. ONNX doesn't support propagating output dict names, while OV does this (not perfectly though). The previous implementation was just checking if the raw model outputs correspond to the parameter passed to the exporter, and clean up the mess introduced by OVC where possible. Now the implementation just assigns the names. I assume we can do that in OV since we have to provide correct output names for ONNX model anyway.
At the same time I see your point, I'll add an extra check to OV implementation to produce warnings if the output names are inconsistent vs ones in the output dict.